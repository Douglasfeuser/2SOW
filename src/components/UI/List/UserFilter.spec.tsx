import React from "react";
import { UserFilter } from "./UserFilter";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

jest.mock("lodash.debounce", () => jest.fn(fn => fn));

describe("UserFilter", () => {
  it("renders", () => {
    const filter = "";
    const totalCount = 100;
    const onSubmitFilter = jest.fn();

    const { container } = render(
      <UserFilter
        filter={filter}
        totalCount={totalCount}
        onSubmitFilter={onSubmitFilter}
      />
    );

    expect(container).toMatchSnapshot();
    expect(onSubmitFilter).not.toHaveBeenCalled();
  });

  it("should show message when 0 total count", () => {
    const filter = "";
    const totalCount = 0;
    const onSubmitFilter = jest.fn();

    render(
      <UserFilter
        filter={filter}
        totalCount={totalCount}
        onSubmitFilter={onSubmitFilter}
      />
    );

    waitFor(() => expect(screen.getByText("results")).toBeInTheDocument());
  });

  it("submits", () => {
    const filter = "";
    const totalCount = 10;
    const onSubmitFilter = jest.fn();

    render(
      <UserFilter
        filter={filter}
        totalCount={totalCount}
        onSubmitFilter={onSubmitFilter}
      />
    );

    expect(onSubmitFilter).not.toHaveBeenCalled();

    userEvent.type(screen.getByRole("textbox"), "Volvo");

    expect(onSubmitFilter).toHaveBeenCalled();
    expect(onSubmitFilter.mock.calls).toHaveLength(5);
  });

  it("submits invalid", () => {
    const filter = "";
    const totalCount = 10;
    const onSubmitFilter = jest.fn();

    render(
      <UserFilter
        filter={filter}
        totalCount={totalCount}
        onSubmitFilter={onSubmitFilter}
      />
    );

    userEvent.type(screen.getByRole("textbox"), "#");

    expect(onSubmitFilter).not.toHaveBeenCalled();
    waitFor(() => expect(screen.getByText("Invalid")).toBeInTheDocument());
  });
});
