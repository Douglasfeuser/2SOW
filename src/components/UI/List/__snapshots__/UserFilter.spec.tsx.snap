// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFilter renders 1`] = `
<div>
  <form
    class="ui form"
  >
    <div
      class="fields"
    >
      <div
        class="field"
      >
        <div
          class="field"
        >
          <label>
            Filter
          </label>
          <div
            class="ui icon input"
          >
            <input
              name="filter"
              placeholder="Pesquisa."
              type="text"
            />
            <i
              aria-hidden="true"
              class="search icon"
            />
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
